### ISSUE TITLE
Collapsible/expandable sequence of small events

### ISSUE THREAD

**Issue by kevinaboos** [project-robius/robrix#118]
Small state events can take up a lot of vertical space on the timeline. This includes things like room membership changes, room metadata changes, profile name/avatar changes, etc. 

By default, a series of two or more adjacent small events should be collapsed into an abbreviated view that takes up much less vertical space. This will make it easier for the user to focus on important content like text and image messages from other users, rather than less important metadata.

For example, this is how Element shows a series of contiguous state events:
<img width="821" alt="image" src="https://github.com/user-attachments/assets/dd0a02f5-8a38-4ff9-839d-60097ed87f4b">

and then when expanded, it looks like what Robrix currently shows:
<img width="817" alt="image" src="https://github.com/user-attachments/assets/332ba795-34a7-4bee-8169-fe3b30f6fc47">

We would also want dynamic buttons that toggle between "expand" and "collapse" based on whether a range of state events was collapsed or expanded.

**Comment by ZhangHanDong** [comment]
Inspired by the recent new UI of Rust doc, I feel that changing "Collapsible/expandable" to "Summary/show all" may better convey the semantics of this feature.

**Comment by kevinaboos** [comment]
Indeed, I agree. It's also related to github's details+summary markdown feature: https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections
